Bugsnag/KSCrash/Source/KSCrash/Recording/Sentry/BSG_KSCrashSentry_CPPException.mm

```diff
@@ -131,15 +130,13 @@ static void CPPExceptionTerminate(void) {
     try {
         throw;
     } catch (NSException *exception) {
-        BSG_KSLOG_DEBUG("Detected NSException. Recording details and letting "
-                        "the current NSException handler deal with it.");
-        isNSException = true;
-        // recordException() doesn't call beginHandlingCrash()
-        if (bsg_kscrashsentry_isNSExceptionHandlerInstalled() &&
-            bsg_kscrashsentry_beginHandlingCrash(bsg_ksmachthread_self())) {
-            bsg_recordException(exception);
-            bsg_kscrashsentry_endHandlingCrash();
+        if (bsg_g_originalTerminateHandler != NULL) {
+            BSG_KSLOG_DEBUG("Detected NSException. Passing to the current NSException handler.");
+            bsg_g_originalTerminateHandler();
```

Comment:
How do NSExceptions now get captured?

Does `bsg_g_originalTerminateHandler` end up calling `bsg_ksnsexc_i_handleException` via `NSUncaughtExceptionHandler`?

Reply:
The ObjC runtime registers a C++ exception handler that checks for NSException and does processing if it is (calling whatever NSUncaughtExceptionHandler is registered, or calling `abort()` if none is).

We tap into both the C++ exception handler and the NSException handler, but since we don't want to process twice, we check for NSException and just pass the exception along for the ObjC runtime to find.

Reply:
What would happen if bsg_g_originalTerminateHandler is (for example) Crashlytics's NSUncaughtExceptionHandler?

Will this PR mean that crashes we previously reported in apps that use both would no longer be detected?

Reply:
Hmm actually, if someone else injects themselves into the callback chain AND forgets to pass it along, we'd get an abort signal instead (if you hook the c++ exception callback and don't pass it along to the next handler, the ObjC runtime will never see it).

So, also capturing information at the C++ level would give an extra defence against poorly implemented third party software (but only if they hook the handler BEFORE we do; if they hook after us and forget to pass it on, all is lost).

Reply:
Nope, my initial testing was flawed, and further testing showed that this C++ code is not capturing NSException data after all. So we're better off going with the approach in this PR to ensure that `bsg_kscrashsentry_beginHandlingCrash` never gets called more than once for a single crash.